lib/ansible/plugins/test/success.yml

```
@@ -5,8 +5,9 @@ DOCUMENTATION:
   short_description: check task success
   aliases: [succeeded, successful]
   description:
-    - Tests if task finished successfully, opposite of C(failed).
+    - Test if task did not fail; returning the opposite of C(failed).
     - This test checks for the existence of a C(failed) key in the input dictionary and that it is V(False) if present
+    - This test does not distinguish between a task that completed successfully and a task that was skipped
```

```suggestion
    - A skipped task is a successful task. See P(ansible.builtin.skipped#test) to test if a task is skipped.
```

---

I was looking at the other issue, and I think this stems from a misunderstanding between tasks and actions. If you run a single action and it's skipped, loop over a task with an empty list, or all of the actions in the loop are skipped, then the task result is `ansible.builtin.skipped`. If you run a single action and it fails, or any action in the loop fails, then the overall task result is failed (and success is the inverse of that, which includes skipped as a subset).

```suggestion
    - To validate if the task's actions were executed, see P(ansible.builtin.skipped#test).
```

---

i find the original better, your change seems redundant

---

Not sure which change that is referring to but I agree that referring to the skipped test is useful, I updated that in line with s-hertel's suggestion. Thanks.